Scripts/datatypes/purpose.py

```diff
@@ -278,10 +279,10 @@ def calc_prob(self, impedance, is_last_iteration):
                 Type (time/cost/dist) : numpy 2d matrix
         """
         purpose_impedance = self.transform_impedance(impedance)
-        self.prob = self.model.calc_prob(purpose_impedance)
         if is_last_iteration and self.name[0] != 's':
             self.accessibility_model.calc_accessibility(
-                purpose_impedance)
+                copy(purpose_impedance))
+        self.prob = self.model.calc_prob(purpose_impedance)
```

Comment:
By copying the dict, we can pop the matrices in them on usage, hence depleting both dicts. When the reference count for the matrices drops to zero (in `calc_prob`), memory is released.